src/api/routes/ws/channels/socket-io-channel.ts

```diff
@@ -153,74 +163,126 @@ export class SocketIOChannel extends WebSocketChannel {
     return false;
   }
 
+  private async getTopicSockets(room: Topic) {
+    if (!this.io) {
+      return;
+    }
+    const sockets = [];
+    const socketIds = await this.io.to(room).allSockets();
+    for (const id of socketIds) {
+      const socket = this.io.sockets.sockets.get(id);
+      if (socket) {
+        sockets.push(socket);
+      }
+    }
+    return sockets;
+  }
+
   send<P extends keyof WebSocketPayload>(
     payload: P,
     ...args: ListenerType<WebSocketPayload[P]>
   ): void {
     if (!this.io) {
       return;
     }
+    // If a client takes more than this number of ms to respond to an event `emit`, it will be
+    // disconnected.
+    const timeout = getWsMessageTimeoutMs();
     switch (payload) {
       case 'block': {
         const [block] = args as ListenerType<WebSocketPayload['block']>;
         this.prometheus?.sendEvent('block');
-        this.io.to('block').emit('block', block);
+        void this.getTopicSockets('block').then(sockets =>
+          sockets?.map(s => s.timeout(timeout).emit('block', block, _ => s.disconnect(true)))
```

Comment:
Is `map` used here for a reason? It seems like the usage is more inline with a `forEach`?

Reply:
Nope, nice catch too, done!